sharding-orchestration/sharding-orchestration-core/src/main/java/io/shardingsphere/orchestration/internal/registry/config/listener/SchemaChangedListener.java

```diff
@@ -0,0 +1,138 @@
+import io.shardingsphere.orchestration.internal.registry.config.event.SchemaAddChangedEvent;
+import io.shardingsphere.orchestration.internal.registry.config.event.SchemaDeleteChangedEvent;
+import io.shardingsphere.orchestration.internal.registry.config.event.ShardingRuleChangedEvent;
+import io.shardingsphere.orchestration.internal.registry.config.node.ConfigurationNode;
+import io.shardingsphere.orchestration.internal.registry.config.service.ConfigurationService;
+import io.shardingsphere.orchestration.internal.registry.listener.PostShardingOrchestrationEventListener;
+import io.shardingsphere.orchestration.internal.registry.listener.ShardingOrchestrationEvent;
+import io.shardingsphere.orchestration.reg.api.RegistryCenter;
+import io.shardingsphere.orchestration.reg.listener.DataChangedEvent;
+import io.shardingsphere.orchestration.reg.listener.DataChangedEvent.ChangedType;
+import io.shardingsphere.orchestration.yaml.ConfigurationYamlConverter;
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+/**
+ * Schema changed listener.
+ *
+ * @author panjuan
+ */
+public final class SchemaChangedListener extends PostShardingOrchestrationEventListener {
+    
+    private final ConfigurationService configurationService;
+    
+    private final ConfigurationNode configurationNode;
+    
+    private final Collection<String> existedSchemaNames = new LinkedList<>();
+    
+    public SchemaChangedListener(final String name, final RegistryCenter regCenter, final Collection<String> shardingSchemaNames) {
+        super(regCenter, new ConfigurationNode(name).getSchemaPath());
+        configurationService = new ConfigurationService(name, regCenter);
+        configurationNode = new ConfigurationNode(name);
+        existedSchemaNames.addAll(shardingSchemaNames);
+    }
+    
+    @Override
+    protected ShardingOrchestrationEvent createShardingOrchestrationEvent(final DataChangedEvent event) {
+        String shardingSchemaName = configurationNode.getSchemaName(event.getKey());
+        if (Strings.isNullOrEmpty(shardingSchemaName) || !isValidNodeChangedEvent(shardingSchemaName, event.getKey())) {
+            return new IgnoredChangedEvent();
+        }
+        if (ChangedType.UPDATED == event.getChangedType()) {
+            return createUpdateChangedEvent(shardingSchemaName, event);
+        }
+        if (ChangedType.DELETED == event.getChangedType()) {
+            return createDeleteChangedEvent(shardingSchemaName);
+        }
+        return new IgnoredChangedEvent();
+    }
+    
+    private boolean isValidNodeChangedEvent(final String shardingSchemaName, final String nodeFullPath) {
+        return configurationNode.getDataSourcePath(shardingSchemaName).equals(nodeFullPath) || configurationNode.getRulePath(shardingSchemaName).equals(nodeFullPath);
+    }
+    
+    private ShardingOrchestrationEvent createUpdateChangedEvent(final String shardingSchemaName, final DataChangedEvent event) {
```

Comment:
How about rename this method as `createUpdatedEvent`?

Reply:
Please refer to other event names, e.g.PropertiesChangedEvent, ShardingRuleChangedEvent.